dnsdist, rec: Replace net-snmp-includes.h with hand-picked headers

- Program: Recursor, dnsdist
 - Issue type: Bug report

### Short description

We currently include `net-snmp-includes.h` from `snmp-agent.hh`, and this in turn pulls a lot of headers, some of them we don't need, including `net-snmp/library/getopt.h`. This last one does break `dnsdist` on `FreeBSD` with:

`dnsdist.cc:1715:13: error: use of undeclared identifier 'required_argument'`

We should hand-pick the headers we actually need instead of `net-snmp-includes.h`.

--- a/pdns/snmp-agent.hh
+++ b/pdns/snmp-agent.hh
@@ -9,7 +9,10 @@
 
 #ifdef HAVE_NET_SNMP
 #include <net-snmp/net-snmp-config.h>
-#include <net-snmp/net-snmp-includes.h>
+#include <net-snmp/definitions.h>
+#include <net-snmp/types.h>
+#include <net-snmp/utilities.h>
+#include <net-snmp/config_api.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
 #undef INET6 /* SRSLY? */
 #endif /* HAVE_NET_SNMP */

